crates/solidity/outputs/rust/lib/src/generated/kinds.rs

```diff
@@ -569,3 +570,172 @@ pub enum RuleKind {
     YulSwitchStatement,
     YulVariableDeclaration,
 }
+
+#[derive(
+    Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, EnumString, AsRefStr, Display,
+)]
+pub enum ProductionKind {
```

Comment:
Thinking out loud. I worry the additional enum (with duplicaten fields) might cause confusion on user side, and make it harder to correlate between `ProductionKind` and the existing two kinds above.

If this is purely for the sake of the `get_parser()` API, thoughts about creating two APIs instead?

```rust
Language::new(args.version).get_rule_parser(RuleKind::Foo);
Language::new(args.version).get_token_parser(TokenKind::Foo);
```

Reply:
No, it's somewhat wider impact than this, but I agree, and want to revisit this after the next few test-blocking issues, to clean up the code in general.

Reply:
[#226] (https://github.com/NomicFoundation/slang/issues/266)

Reply:
[#267](https://github.com/NomicFoundation/slang/issues/267)